src/OpenTracing/Util/GlobalTracer.cs

```diff
@@ -82,12 +98,16 @@ public static void Register(ITracer tracer)
             lock (s_lock)
             {
                 if (tracer == s_instance._tracer)
+                {
+                    _isRegistered = true;
```

Comment:
Any reason why this is required?  The s_instance._tracer & _isRegistered assignation below works atomically, no?

Reply:
We need this because the default tracer is a NoopTracer, so if you intentionally set a NoopTracer we need to set the s_isRegistered bool to true too.

Reply:
Right, right. Had forgotten this detail ;)